PR : Faz2 General Fixes

anilhazar opened the pull request.

anilhazar committed:
Remove LocalDateConverter Class

anilhazar committed:
Endpoint Fixes for Controller Classes

anilhazar committed:
Make Converter Classes Final and Add Private Constructor

anilhazar committed:
Organize Request Packages and Setter/Getters

anilhazar committed:
Organize Response Getter/Getters

anilhazar committed:
Organize LeaveRequestEntity Fields

anilhazar committed:
Organize Repository Packages

anilhazar committed:
Organize Repository/Impl Packages

anilhazar committed:
Update Generate Random Username Algorithm

anilhazar committed:
Update PasswordEncode System and Use BCryptPasswordEncoder Library

anilhazar committed:
Add EncoderConfig and gitignore

anilhazar committed:
Update .gitignore

anilhazar committed:
Add @Valid Annotation

anilhazar committed:
Request Status Update Endpoint

anilhazar committed:
List Today's Leaves

anilhazar committed:
General Syntax Fixes

anilhazar committed:
Add Status Change Update

anilhazar committed:
General Naming Fixes

anilhazar committed:
Add Optional as a Feature and Naming Fixes

anilhazar committed:
Implementation of "List Leaves By Status" API Endpoint:
Related endpoint's end to end development with Status Filter and Pagination completed successfully.

agitrubard commented on pom.xml:
```diff
@@ -37,6 +37,12 @@
+		<dependency>
+			<groupId>org.springframework.security</groupId>
+			<artifactId>spring-security-crypto</artifactId>
+			<version>6.1.4</version>
```
Versiyon kontrollerini kolay bir şekilde yapabilmek için versiyonu `pom.xml` içerisindeki properties kısmına tanımlayabilirsin

anilhazar replied:
Geliştirme yapıldı

anilhazar committed:
Add "birthday" as an employee table column and overall implementation of birthday field to project

anilhazar committed:
Implement daily birthday email notifications:

Development of the functionality to send birthday emails to users every day at 9 AM completed successfully

anilhazar committed:
General Fixes

anilhazar committed:
General Fixes

anilhazar committed:
Add pagination and Filter feats to Get Leave Requests By Status Endpoint

anilhazar committed:
Fix/Improve Exception Handling of ServiceImpl Classes

anilhazar committed:
Fix/Improve Code Syntax and List Exceptions

anilhazar committed:
Refactor the sendBirthdayEmail method in EmployeeService

anilhazar committed:
sql.script update

anilhazar committed:
refactor sql script

agitrubard commented on src/main/resources/sql/hr-be-api.sql:
```diff
@@ -11,26 +11,24 @@ CREATE TABLE employee
 (
     id           BIGINT AUTO_INCREMENT PRIMARY KEY               NOT NULL,
     employee_id  BIGINT                                          NOT NULL,
     request_type ENUM ('BIRTH', 'ANNUAL', 'PATERNITY', 'EXCUSE') NOT NULL,
```
```suggestion
    type ENUM ('BIRTH', 'ANNUAL', 'PATERNITY', 'EXCUSE') NOT NULL,
```

anilhazar committed:
Scheduled annotation fix

anilhazar committed:
Refactor name of request_type column of leave_request table

Co-authored-by: Agit Rubar Demir <61833677+agitrubard@users.noreply.github.com>

anilhazar committed:
Add spring security crypto version as a property

agitrubard commented on pom.xml:
```diff
@@ -18,6 +18,7 @@
 		<java.version>17</java.version>
 		<sql2o.version>1.8.0</sql2o.version>
         <mysql.connector.version>8.3.0</mysql.connector.version>
```
Bu dosyanın formatında problem var sanırım

anilhazar replied:
kontrol ettim bir sorun görünmüyor

agitrubard commented on src/main/java/com/spring/model/dto/request/EmployeeCreateRequest.java:
```diff
@@ -1,62 +1,78 @@
+    private LocalDate birthday;
+
     @Positive
     private int age;
```
```suggestion
    private Integer age;
```

anilhazar replied:
Done

agitrubard commented on src/main/java/com/spring/model/dto/response/LeaveRequestResponse.java:
```diff
@@ -1,23 +1,43 @@
 
     private final RequestType requestType;
-    private final Status status;
+    private final RequestStatus requestStatus;
```
```suggestion
    private final RequestStatus status;
```

agitrubard commented on src/main/java/com/spring/service/impl/EmployeeEmailServiceImpl.java:
```diff
@@ -27,4 +34,39 @@ public void sendUsernameAndPassword(EmployeeEntity employeeEntity) {
+
+
+    @Scheduled(cron = "0 0 9 * * *")
+    public void sendBirthdayEmail() {
```
Bu metot burada olmasa daha iyi olabilir ve public yaptığın için diğer yerlerden çağrılabiliyor, buna dikkat etmek lazım

anilhazar replied:
Access modifier düzeltildi. Employee Email service sınıfı, bu metodun yaptığı işe uygun diye düşünüyorum.

agitrubard commented on src/main/resources/sql/hr-be-api.sql:
```diff
@@ -1,23 +1,34 @@
+CREATE DATABASE IF NOT EXISTS hr_be_api;
+
+USE hr_be_api;
+
```
Burada bir database şifresi verip bir şema oluşturmak lazım, bu şemanın application.yml'daki default verilerle çalışabiliyor olması lazım

anilhazar replied:
gerekli düzenlemeler yapıldı

anilhazar committed:
Update src/main/java/com/spring/model/dto/request/EmployeeCreateRequest.java

Co-authored-by: Agit Rubar Demir <61833677+agitrubard@users.noreply.github.com>

anilhazar committed:
Update src/main/java/com/spring/model/dto/response/LeaveRequestResponse.java

Co-authored-by: Agit Rubar Demir <61833677+agitrubard@users.noreply.github.com>

anilhazar committed:
sql script, app.yml and sendBirthdayEmail method refactored,

anilhazar committed:
feat: Add unique email constraint in service and database

- Implemented a check in the service layer to ensure that no two employees can have the same email address.
- Added a unique constraint to the email column in the SQL script to enforce uniqueness at the database level.
- Used `Optional<EmployeeEntity>` in the service method to streamline the check for existing emails.
- Refactored method names in service and repository layers to avoid redundant naming complexities.

anilhazar committed:
Merge remote-tracking branch 'origin/fix/general-fixes' into fix/general-fixes

anilhazar committed:
feat: Add oldPassword check in changePassword method

- Implemented a password check in the changePassword method to ensure that users password matches with the input.
- Added oldPassword field in EmployeePasswordChangeRequest with relative constraints.

anilhazar committed:
fix: Send raw password in email instead of hashed password

- Refactored parameters of sendUsernameAndPassword method.

- Refactored EmployeeRepositoryConstants fields.

- Refactored update method in EmployeeRepositoryImpl

anilhazar committed:
fix: Refactor enums names

- Refactored RequestType enum to Type in whole project.

- Refactored RequestStatus enum to Status in whole project.

agitrubard commented on src/main/resources/application.yml:
```diff
@@ -8,8 +8,8 @@ spring:
-    username:
-    password:
+    username: ${db.username:name}
+    password: ${db.password:password}
```
Buradaki `name` ve `password` ile veri tabanına bağlanılabiliyor mu?

anilhazar committed:
fix: Refactor app.yml

- Refactored app.yml values.

- Fixed general syntax errors.
